docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerLauncher.java

```diff
@@ -0,0 +1,73 @@
+package com.nirima.jenkins.plugins.docker.launcher;
+
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.nirima.jenkins.plugins.docker.DockerTemplate;
+import com.nirima.jenkins.plugins.docker.DockerTemplateBase;
+import hudson.model.TaskListener;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.DelegatingComputerLauncher;
+import hudson.slaves.SlaveComputer;
+
+import java.io.IOException;
+
+
+/**
+ * Crappy wrapper... On one hand we need store UI configuration,
+ * on other have valid configured launcher that different for host/port/etc for any slave.
+ * <p/>
+ * like {@link DelegatingComputerLauncher}
+ */
+public abstract class DockerComputerLauncher extends ComputerLauncher {
+    protected ComputerLauncher launcher;
+
+    /**
+     * Return valid configured launcher that will be used for launching slave
+     */
+    public abstract ComputerLauncher getPreparedLauncher(String cloudId, DockerTemplate dockerTemplate, InspectContainerResponse ir);
+
+    /**
+     * Contribute container parameters needed for launcher.
+     * i.e. port for exposing, command to run, etc.
+     */
+    public abstract void appendContainerConfig(DockerTemplateBase dockerTemplate, CreateContainerCmd createContainerCmd);
```

Comment:
Not a problem, can be changed later. I don't expect that somebody will implement this outside this plugin.